js/compiler_tests.ts

```diff
@@ -0,0 +1,466 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import { test, assert, assertEqual } from "./testing/testing.ts";
+import * as compiler from "compiler";
+import * as ts from "typescript";
+
+// We use a silly amount of `any` in these tests...
+// tslint:disable:no-any
+
+const { DenoCompiler, ModuleMetaData } = compiler;
+
+// Enums like this don't exist at runtime, so local copy
+enum ScriptKind {
+  JS = 1,
+  TS = 3,
+  JSON = 6
+}
+
+interface ModuleInfo {
+  moduleName: string | null;
+  filename: string | null;
+  sourceCode: string | null;
+  outputCode: string | null;
+}
+
+const compilerInstance = DenoCompiler.instance();
+
+// References to orignal items we are going to mock
+const originals = {
+  _globalEval: (compilerInstance as any)._globalEval,
+  _log: (compilerInstance as any)._log,
+  _os: (compilerInstance as any)._os,
+  _ts: (compilerInstance as any)._ts,
+  _service: (compilerInstance as any)._service,
+  _window: (compilerInstance as any)._window
+};
+
+function mockModuleInfo(
+  moduleName: string | null,
+  filename: string | null,
+  sourceCode: string | null,
+  outputCode: string | null
+): ModuleInfo {
+  return {
+    moduleName,
+    filename,
+    sourceCode,
+    outputCode
+  };
+}
+
+// Some fixtures we will us in testing
+const fooBarTsSource = `import * as compiler from "compiler";
+console.log(compiler);
+export const foo = "bar";
+`;
+
+const fooBazTsSource = `import { foo } from "./bar.ts";
+console.log(foo);
+`;
+
+// tslint:disable:max-line-length
+const fooBarTsOutput = `define(["require", "exports", "compiler"], function (require, exports, compiler) {
+    "use strict";
+    Object.defineProperty(exports, "__esModule", { value: true });
+    console.log(compiler);
+    exports.foo = "bar";
+});
+//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiYmFyLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiZmlsZTovLy9yb290L3Byb2plY3QvZm9vL2Jhci50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7SUFDQSxPQUFPLENBQUMsR0FBRyxDQUFDLFFBQVEsQ0FBQyxDQUFDO0lBQ1QsUUFBQSxHQUFHLEdBQUcsS0FBSyxDQUFDIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICogYXMgY29tcGlsZXIgZnJvbSBcImNvbXBpbGVyXCI7XG5jb25zb2xlLmxvZyhjb21waWxlcik7XG5leHBvcnQgY29uc3QgZm9vID0gXCJiYXJcIjtcbiJdfQ==
+//# sourceURL=/root/project/foo/bar.ts`;
```

Comment:
If possible, it would be nice to not have the source map stuff in here. Maybe that needs to wait until we disable the inlineSourceMaps - if so please add TODO
```
// TODO Remove source map strings from fooBarTsOutput. https://github.com/denoland/deno/issues/23
```

Reply:
Done